client/http/client.go

```diff
@@ -181,6 +183,8 @@ func (ci *clientInner) doRequest(
 		zap.String("caller-id", callerID),
 	}
 	log.Debug("[pd] request the http url", logFields...)
+	ctx, cancel = context.WithTimeout(ctx, reqInfo.timeout)
```

Comment:
Set a timeout for each request may not be a good idea. How about just supporting only one way to set timeout: init timeout in client with WithTimeout function, and timeout mechanism also through Timeout in http.Client?

Reply:
Got it. My initial idea was to create as few http.clients as possible, but I've found that this is not a good way to modify it.